File: apps/rails/app/models/icon_embedding.rb
# == Schema Information
#
# Table name: icon_embeddings
#
#  id         :integer          not null, primary key
#  key        :string           not null
#  text       :text             not null
#  embedding  :vector(1536)     not null
#  metadata   :jsonb            default("{}"), not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
# Indexes
#
#  idx_icon_embeddings_text      (embedding)
#  index_icon_embeddings_on_key  (key) UNIQUE
#

class IconEmbedding < ApplicationRecord
end

File: apps/langgraph/db/schema/iconEmbedding.ts
import { pgTable, serial, text, timestamp, jsonb, index } from 'drizzle-orm/pg-core';
import { vector } from 'drizzle-orm/pg-core';
import { createInsertSchema } from 'drizzle-zod';
import { z } from 'zod';

export const iconEmbedding = pgTable('icon_embeddings', {
  id: serial('id').primaryKey(),
  key: text('key').notNull().unique(),
  text: text('text').notNull(),
  embedding: vector('embedding', { dimensions: 1536 }),
  metadata: jsonb('metadata'),
  createdAt: timestamp('created_at').defaultNow().notNull()
}, (table) => ({
  vectorIdx: index('icon_embedding_idx')
    .using('ivfflat', table.embedding.op('vector_cosine_ops'))
    .with({ lists: 100 }),
  textIdx: index('icon_text_idx').on(table.text)
}));

export const iconEmbeddingSchema = createInsertSchema(iconEmbedding);
export type IconEmbeddingSchema = z.infer<typeof iconEmbeddingSchema>;